faunadb-java-dsl/src/main/java/com/faunadb/client/types/Field.java

```diff
@@ -120,6 +131,26 @@ private Field(Path path, Codec<T> codec) {
   }
 
   /**
+   * Creates a field extractor that coerces its value to the class passed
+   *
+   * @param type class to be used to coerce the field's value
+   * @return a new field that coerces its value using the class passed
+   */
+  public <A> Field<A> to(final Class<A> type) {
+    return new Field<>(path, new Codec<A>() {
+      @Override
+      public Result<A> decode(Value value) {
+        return Decoder.decode(value, type);
+      }
+
+      @Override
+      public Result<Value> encode(A value) {
+        return Encoder.encode(value);
+      }
+    });
+  }
+
+  /**
    * Creates a field extractor that collects each inner value of an array using the nested field passed,
    * assuming the root value is an instance of {@link com.faunadb.client.types.Value.ArrayV}
    *
```

Comment:
"nested field" => "nested {@link Field}" 

Would "target value" be better than "root value"?